src/plots/xyPlot.ts

```diff
@@ -22,6 +35,95 @@ export class XYPlot<X, Y> extends Plot {
+  }
+
+  private _fastPanZoomOnX(scale: Scale<X, number>) {
+    if (!this._isAnchored) {
+      return;
+    }
+
+    var domain = scale.domain();
+    this._fastPanZoomScaleX = (scale.scale(this._fastPanZoomKnownDomainX[1]) - scale.scale(this._fastPanZoomKnownDomainX[0])) /
+                              (scale.scale(domain[1]) - scale.scale(domain[0]));
+    this._fastPanZoomDeltaX = scale.scale(this._fastPanZoomKnownDomainX[0]) - scale.scale(domain[0]);
+
+    if (this._renderArea != null) {
+      this._renderArea.attr("transform",
+        "translate(" + this._fastPanZoomDeltaX + ", " + this._fastPanZoomDeltaY + ")" +
+        "scale(" + this._fastPanZoomScaleX + ", " + this._fastPanZoomScaleY + ")");
+      clearTimeout(this._fastPanZoomTimeoutReferenceX);
+      this._fastPanZoomTimeoutReferenceX = setTimeout(() => {
+        this._fastPanZoomKnownDomainX = domain;
+        this._fastPanZoomDeltaX = 0;
+        this._fastPanZoomDeltaY = 0;
+        this.render();
+        this._renderArea.attr("transform", "translate(0, 0) scale(1, 1)");
+      }, 500);
+    }
+
+  }
+
+  private _fastPanZoomOnY(scale: Scale<Y, number>) {
+    if (!this._isAnchored) {
+      return;
+    }
+
+    var domain = scale.domain();
+    this._fastPanZoomScaleY = (scale.scale(this._fastPanZoomKnownDomainY[1]) - scale.scale(this._fastPanZoomKnownDomainY[0])) /
+                  (scale.scale(domain[1]) - scale.scale(domain[0]));
+
+    this._fastPanZoomDeltaY = scale.scale(this._fastPanZoomKnownDomainY[0]) - scale.scale(domain[0]) * this._fastPanZoomScaleY;
+
+    if (!this._renderArea != null) {
+      this._renderArea.attr("transform",
+        "translate(" + this._fastPanZoomDeltaX + ", " + this._fastPanZoomDeltaY + ")" +
+        "scale(" + this._fastPanZoomScaleX + ", " + this._fastPanZoomScaleY + ")");
+      clearTimeout(this._fastPanZoomTimeoutReferenceY);
+      this._fastPanZoomTimeoutReferenceY = setTimeout(() => {
+        this._fastPanZoomKnownDomainY = domain;
+        this._fastPanZoomDeltaX = 0;
+        this._fastPanZoomDeltaY = 0;
+        this.render();
+        this._renderArea.attr("transform", "translate(0, 0) scale(1, 1)");
+      }, 500);
+    }
+  }
+
+  public performanceEnabled(): boolean;
+  public performanceEnabled(performanceEnabled: boolean): XYPlot<X, Y>;
+  public performanceEnabled(performanceEnabled?: boolean): any {
+    if (performanceEnabled == null) {
+      return this._performanceEnabled;
+    }
+
+    if (performanceEnabled) {
+      if (this.x() && this.x().scale) {
+        this._fastPanZoomKnownDomainX = this.x().scale.domain();
+        this.x().scale.onUpdate(this._fastPanZoomOnXCallback);
+        this.x().scale.offUpdate(this._renderCallback);
+      }
+      if (this.y() && this.y().scale) {
+        this._fastPanZoomKnownDomainY = this.y().scale.domain();
+        this.y().scale.onUpdate(this._fastPanZoomOnYCallback);
+        this.y().scale.offUpdate(this._renderCallback);
+      }
```

Comment:
Fixed. It failed because other (now) known bugs https://github.com/palantir/plottable/issues/2396